src/v/cloud_storage/remote.cc

```diff
@@ -344,7 +344,6 @@ ss::future<upload_result> remote::upload_segment(
         if (res) {
             _probe.successful_upload();
             _probe.register_upload_size(content_length);
-            co_await reader_handle->close();
```

Comment:
for reviewers you need to look up above further in the file to see the redundancy